### ISSUE TITLE
driver: add couchbase support

### ISSUE THREAD

**Issue by Dentrax** [k3s-io/kine#197]
It'd be fun and easy to implement. If the maintainers say OK, it's ready to go.

Library is: https://github.com/couchbase/gocb

Dropping the idea here as _good-first-issue_ so we don't forget.

**Comment by brandond** [comment]
I think this would probably be fairly non-trivial to implement this, as couchbase is more of a document-oriented database and isn't a good fit for the row-based semantics that Kine's sql backends expect. It might be possible to implement a more direct backend, similar to what was done for NATS Jetstream, but it would probably have to come from someone with a better understanding of couchbase then our team currently has available.

in short - I don't think we're going to do it ourselves, but we'd welcome a PR from the community.
